lsp-types/src/Language/LSP/Types/Capabilities.hs

```diff
@@ -1,19 +1,24 @@
-{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE OverloadedLabels      #-}
+{-# LANGUAGE OverloadedStrings     #-}
 module Language.LSP.Types.Capabilities
   (
-    module Language.LSP.Types.ClientCapabilities
-  , module Language.LSP.Types.ServerCapabilities
-  , module Language.LSP.Types.WorkspaceEdit
-  , fullCaps
+    fullCaps
   , LSPVersion(..)
   , capsForVersion
   ) where
 
-import Prelude hiding (min)
-import Language.LSP.Types.ClientCapabilities
-import Language.LSP.Types.ServerCapabilities
-import Language.LSP.Types.WorkspaceEdit
-import Language.LSP.Types
+import           Data.Row
+import           Language.LSP.Types.Common
+import           Language.LSP.Types.CodeAction
+import           Language.LSP.Types.SemanticTokens
+import           Language.LSP.Types.Internal.Generated
+import           Prelude                               hiding (min)
+
+{-
+TODO: this is out-of-date/needs an audit
+TODO: can we generate this? process the 'since' annotations in the metamodel?
```

Comment:
Seems like a good idea to generate this as well. I'm going cross-eyed looking at it. I can give it a shot if you want